Project/Source/Modules/ModuleInput.cpp

```diff
@@ -221,4 +282,43 @@ KeyState* ModuleInput::GetMouseButtons() {
 
 KeyState* ModuleInput::GetKeyboard() {
 	return keyboard;
-}
\ No newline at end of file
+}
+
+void ModuleInput::OnControllerAdded(int index) {
+	LOG("Added controller");
+	if (SDL_IsGameController(index)) {
+		LOG("Found controller as SDL Game Controller");
+
+		if (playerControllers[0] == nullptr) {
+			LOG("New controller took player slot 0");
+			playerControllers[0] = new PlayerController(index);
+		} else if (playerControllers[1] == nullptr) {
+			LOG("New controller took player slot 1");
+			playerControllers[1] = new PlayerController(index);
+		}
+	}
+}
+
+void ModuleInput::OnControllerRemoved(int index) {
+	LOG("Disconnected controller");
+	for (int i = 0; i < MAX_PLAYERS; i++) {
+		if (playerControllers[i] == nullptr) continue;
+		if (playerControllers[i]->joystickIndex == index) {
+			RELEASE(playerControllers[i]);
+			LOG("Player slot %d is now empty", i);
+		}
+	}
+}
+
+PlayerController* ModuleInput::GetPlayerControllerWithJoystickIndex(int joystickIndex) const {
+	for (int i = 0; i < MAX_PLAYERS; i++) {
+		if (playerControllers[i] == nullptr) continue;
+		if (playerControllers[i]->joystickIndex == joystickIndex) {
+			return playerControllers[i];
+		}
+	}
+}
```

Comment:
Warning: Not all control paths return a value

Reply:
True!